src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/GregtechMetaTileEntity_IndustrialChisel.java

```diff
@@ -0,0 +1,438 @@
+
+			GT_Recipe tRecipe = generateChiselRecipe(aItems.get(0), this.getGUIItemStack());
+			
+			if (tRecipe == null) {
+				log("BAD RETURN - 0");
+				return false;	
+			}
+
+			// Based on the Processing Array. A bit overkill, but very flexible.	
+			ItemStack[] aItemInputs = aItems.toArray(new ItemStack[aItems.size()]);	
+			FluidStack[] aFluidInputs = new FluidStack[] {};	
+
+			// Reset outputs and progress stats
+			this.mEUt = 0;
+			this.mMaxProgresstime = 0;
+			this.mOutputItems = new ItemStack[]{};
+			this.mOutputFluids = new FluidStack[]{};
+
+			long tVoltage = getMaxInputVoltage();
+			byte tTier = (byte) Math.max(1, GT_Utility.getTier(tVoltage));
+			long tEnergy = getMaxInputEnergy();
+			log("Running checkRecipeGeneric(0)");	
+
+			log("Running checkRecipeGeneric(1)");
+			// Remember last recipe - an optimization for findRecipe()
+			this.mLastRecipe = tRecipe;
+
+			int aMaxParallelRecipes = getMaxParallelRecipes();
+			int aEUPercent = getEuDiscountForParallelism();
+			int aSpeedBonusPercent = 200;
+
+			aMaxParallelRecipes = this.canBufferOutputs(tRecipe, aMaxParallelRecipes);
+			if (aMaxParallelRecipes == 0) {
+				log("BAD RETURN - 2");
+				return false;
+			}
+
+			// EU discount
+			float tRecipeEUt = (tRecipe.mEUt * aEUPercent) / 100.0f;
+			float tTotalEUt = 0.0f;
+
+			int parallelRecipes = 0;
+
+			log("parallelRecipes: "+parallelRecipes);
+			log("aMaxParallelRecipes: "+aMaxParallelRecipes);
+			log("tTotalEUt: "+tTotalEUt);
+			log("tVoltage: "+tVoltage);
+			log("tRecipeEUt: "+tRecipeEUt);
+			// Count recipes to do in parallel, consuming input items and fluids and considering input voltage limits
+			for (; parallelRecipes < aMaxParallelRecipes && tTotalEUt < (tEnergy - tRecipeEUt); parallelRecipes++) {
+				if (!tRecipe.isRecipeInputEqual(true, aFluidInputs, aItemInputs)) {
+					log("Broke at "+parallelRecipes+".");
+					break;
+				}
+				log("Bumped EU from "+tTotalEUt+" to "+(tTotalEUt+tRecipeEUt)+".");
+				tTotalEUt += tRecipeEUt;
+			}
+
+			if (parallelRecipes == 0) {
+				log("BAD RETURN - 3");
+				return false;
+			}
+
+			// -- Try not to fail after this point - inputs have already been consumed! --
+			log("parallelRecipes: "+parallelRecipes);
+			log("aMaxParallelRecipes: "+aMaxParallelRecipes);
+			log("tTotalEUt: "+tTotalEUt);
+			log("tVoltage: "+tVoltage);
+			log("tRecipeEUt: "+tRecipeEUt);
+
+
+			// Convert speed bonus to duration multiplier
+			// e.g. 100% speed bonus = 200% speed = 100%/200% = 50% recipe duration.
+			aSpeedBonusPercent = Math.max(-99, aSpeedBonusPercent);
+			float tTimeFactor = 100.0f / (100.0f + aSpeedBonusPercent);
+			this.mMaxProgresstime = (int)(tRecipe.mDuration * tTimeFactor);
+
+			this.mEUt = (int)Math.ceil(tTotalEUt);
+
+			this.mEfficiency = (10000 - (getIdealStatus() - getRepairStatus()) * 1000);
+			this.mEfficiencyIncrease = 10000;		
+
+			// Overclock
+			if (this.mEUt <= 16) {
+				this.mEUt = (this.mEUt * (1 << tTier - 1) * (1 << tTier - 1));
```

Comment:
That’s how the base multi does it, I just copied the checkRecipe and changed it to suit both this and RB.

Reply:
Guess this is fine then. It probably won't create absurd mEUt or mMaxProgresstime. Still, I cannot locate the base you copied from in GT5.

Reply:
My base multi which is based on the PA. whether or not it’s been refactored in the PA since I can’t say. But afaik it works as intended for all my multis.

Reply:
Oh, it's that gigantic `checkRecipeGeneric`... then it'd be totally fine.